### PR TITLE
Cycpptest

### PR DESCRIPTION
**Pull request by gidden**
cycpp_tests.h was not being used in tests, this now uses it

### CODE REVIEW

**Comment by gidden** [comment]
ping

File: tests/cycpp_tests.py [hunk 1]
```diff
@@ -1040,6 +1042,12 @@ def test_nuclide_uitype():
...
     yield assert_raises, TypeError, f.impl
 
-
+def test_integration():
+    inf = os.path.join(os.path.dirname(__file__), 'cycpp_tests.h')
+    outf = tempfile.NamedTemporaryFile()
+    cmd = 'cycpp.py {} -o {}'.format(inf, outf.name)
+    p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True)
+    assert_equal('', p.stdout.read().decode())
+
```
Comment (mbmcgarry) [inline tests/cycpp_tests.py]: Should the tempfile be closed/removed?

**Comment by mbmcgarry** [comment]
This PR fails nosetests on Yosemite 

```
======================================================================
FAIL: cycpp_tests.test_integration
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mbmcgarry/git/cyclus/tests/cycpp_tests.py", line 1050, in test_integration
    assert_equal('', p.stdout.read().decode())
AssertionError: '' != u'In file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:72:\nIn file included from /usr/include/Availability.h:153:\n/usr/include/AvailabilityInternal.h:5858:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_9\n         ^\n/usr/include/AvailabilityInternal.h:5819:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_8\n         ^\n/usr/include/AvailabilityInternal.h:5784:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_7\n         ^\n/usr/include/AvailabilityInternal.h:5753:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_6\n         ^\n/usr/include/AvailabilityInternal.h:5726:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_5\n         ^\n/usr/include/AvailabilityInternal.h:5703:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_4\n         ^\n/usr/include/AvailabilityInternal.h:5684:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_3\n         ^\n/usr/include/AvailabilityInternal.h:5669:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_2\n         ^\n/usr/include/AvailabilityInternal.h:5658:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_1\n         ^\n/usr/include/AvailabilityInternal.h:58:2: error: unterminated conditional directive\n#ifdef __IPHONE_OS_VERSION_MIN_REQUIRED\n ^\n/usr/include/AvailabilityInternal.h:30:2: error: unterminated conditional directive\n#ifndef __AVAILABILITY_INTERNAL__\n ^\nIn file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:72:\n/usr/include/Availability.h:199:2: error: #else without #if\n#else\n ^\n/usr/include/Availability.h:202:2: error: #endif without #if\n#endif\n ^\n/usr/include/Availability.h:208:2: error: #endif without #if\n#endif /* __AVAILABILITY__ */\n ^\nIn file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:90:\nIn file included from /usr/include/stdio.h:65:\n/usr/include/Availability.h:199:2: error: #else without #if\n#else\n ^\n/usr/include/Availability.h:202:2: error: #endif without #if\n#endif\n ^\n/usr/include/Availability.h:208:2: error: #endif without #if\n#endif /* __AVAILABILITY__ */\n ^\nIn file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:90:\nIn file included from /usr/include/stdio.h:75:\nIn file included from /usr/include/sys/stdio.h:35:\n/usr/include/Availability.h:199:2: error: #else without #if\n#else\n ^\n/usr/include/Availability.h:202:2: error: #endif without #if\n#endif\n ^\nfatal error: too many errors emitted, stopping now [-ferror-limit=]\n20 errors generated.\n'
````

**Comment by gidden** [comment]
can you try running cycpp as this test does to confirm the error?

On Wed, Jun 3, 2015 at 12:41 PM, mbmcgarry <notifications@github.com> wrote:

> This PR fails nosetests on Yosemite
>
> ======================================================================
> FAIL: cycpp_tests.test_integration
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/Library/Python/2.7/site-packages/nose/case.py", line 197, in runTest
>     self.test(*self.arg)
>   File "/Users/mbmcgarry/git/cyclus/tests/cycpp_tests.py", line 1050, in test_integration
>     assert_equal('', p.stdout.read().decode())
> AssertionError: '' != u'In file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:72:\nIn file included from /usr/include/Availability.h:153:\n/usr/include/AvailabilityInternal.h:5858:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_9\n         ^\n/usr/include/AvailabilityInternal.h:5819:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_8\n         ^\n/usr/include/AvailabilityInternal.h:5784:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_7\n         ^\n/usr/include/AvailabilityIntern
>  al.h:5753:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_6\n         ^\n/usr/include/AvailabilityInternal.h:5726:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_5\n         ^\n/usr/include/AvailabilityInternal.h:5703:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_4\n         ^\n/usr/include/AvailabilityInternal.h:5684:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_3\n         ^\n/usr/include/AvailabilityInternal.h:5669:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_2\n         ^\n/usr/include/AvailabilityInternal.h:5658:10: error: unterminated conditional directive\n        #if __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_1\n         ^\n/usr/include/AvailabilityInternal.h:58:2: error: untermi
>  nated conditional directive\n#ifdef __IPHONE_OS_VERSION_MIN_REQUIRED\n ^\n/usr/include/AvailabilityInternal.h:30:2: error: unterminated conditional directive\n#ifndef __AVAILABILITY_INTERNAL__\n ^\nIn file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:72:\n/usr/include/Availability.h:199:2: error: #else without #if\n#else\n ^\n/usr/include/Availability.h:202:2: error: #endif without #if\n#endif\n ^\n/usr/include/Availability.h:208:2: error: #endif without #if\n#endif /* __AVAILABILITY__ */\n ^\nIn file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/Xcode
>  Default.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:90:\nIn file included from /usr/include/stdio.h:65:\n/usr/include/Availability.h:199:2: error: #else without #if\n#else\n ^\n/usr/include/Availability.h:202:2: error: #endif without #if\n#endif\n ^\n/usr/include/Availability.h:208:2: error: #endif without #if\n#endif /* __AVAILABILITY__ */\n ^\nIn file included from /Users/mbmcgarry/git/cyclus/tests/cycpp_tests.h:1:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:435:\nIn file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:90:\nIn file included from /usr/include/wchar.h:90:\nIn file included from /usr/include/stdio.h:75:\nIn file inc
>  luded from /usr/include/sys/stdio.h:35:\n/usr/include/Availability.h:199:2: error: #else without #if\n#else\n ^\n/usr/include/Availability.h:202:2: error: #endif without #if\n#endif\n ^\nfatal error: too many errors emitted, stopping now [-ferror-limit=]\n20 errors generated.\n'
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1174#issuecomment-108541894>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

File: tests/cycpp_tests.py (hunk 1 above)
Comment (gidden) [inline tests/cycpp_tests.py]: The file is closed when garbage collected.

On Wed, Jun 3, 2015 at 12:38 PM, mbmcgarry <notifications@github.com> wrote:

> In tests/cycpp_tests.py
> <https://github.com/cyclus/cyclus/pull/1174#discussion_r31648592>:
>
> > @@ -1040,6 +1042,12 @@ def test_nuclide_uitype():
> >      f.given_classname = 'MyFactory'
> >      yield assert_raises, TypeError, f.impl
> >
> > -
> > +def test_integration():
> > +    inf = os.path.join(os.path.dirname(__file__), 'cycpp_tests.h')
> > +    outf = tempfile.NamedTemporaryFile()
> > +    cmd = 'cycpp.py {} -o {}'.format(inf, outf.name)
> > +    p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True)
> > +    assert_equal('', p.stdout.read().decode())
> > +
>
> Should the tempfile be closed/removed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1174/files#r31648592>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by mbmcgarry** [comment]
Yes, it also fails when I run it directly.  Here's a gist with the beginning of the output including the error (it was very very very long)

https://gist.github.com/mbmcgarry/ab8be8733c7982f467d2

**Comment by gidden** [comment]
@mbmcgary based on this:
https://github.com/ThrowTheSwitch/Ceedling/issues/37 can you try changing
line 1048 from
cmd = 'cycpp.py {} -o {}'.format(inf, outf.name)
to
cmd = 'cycpp.py {} -o {} --cpp-path `which g++`'.format(inf, outf.name)
?

On Wed, Jun 3, 2015 at 12:51 PM, mbmcgarry <notifications@github.com> wrote:

> Yes, it also fails when I run it directly. Here's a gist with the
> beginning of the output including the error (it was very very very long)
>
> https://gist.github.com/mbmcgarry/ab8be8733c7982f467d2
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1174#issuecomment-108545678>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by mbmcgarry** [comment]
Yes!  Your fix works! I can build, run cyclus_unit_tests and run nosetests.

**Comment by gidden** [comment]
fixed

**Comment by gidden** [comment]
rebased

**Comment by mbmcgarry** [comment]
Thanks @gidden , especially for making this Mac-compatible!

**Pull request merged**